torch/quantization/fx/prepare.py

```diff
@@ -812,7 +840,8 @@ def insert_observers_for_model(
                     maybe_insert_input_observers_for_node(
                         node, qconfig, model, modules, graph,
                         node_name_to_target_dtype,
-                        qhandler, prepare_custom_config_dict)
+                        qhandler, prepare_custom_config_dict,
+                        equalization_qconfig)
```

Comment:
the logic for inserting observers for equalization seems a lot simpler (at least at the start).  Can we consider keeping this implenetation fully separate?  For example, we could have:

```
for node in nodes_before_observation:
  ...
  # insert quantization input observers, if needed
  # insert equalization input observers, if needed
  # insert quantization output observers, if needed
```

That way the code for equalization is separate in its own functions, instead of being mixed with the quantization code.